### PR TITLE
V15 components 2.0

### PR DESCRIPTION
**Pull request by booc0mtaco**
## The BIG Merge

- Bring in all the component, documentation, and other changes from the long-lived v15 branch
- Update documentation and deprecation markers throughout
- Adjustments to tokens and corresponding (sub-)component usages

### CODE REVIEW

File: .storybook/preview.tsx [hunk 1]
```diff
@@ -11,61 +11,81 @@ import '../src/design-tokens/tier-1-definitions/fonts.css';
...
-    ...createPaperReleaseConfig('1.2'),
-    ...createPaperReleaseConfig('1.1'),
-    ...createPaperReleaseConfig('1.0'),
+    ...createInitialReleaseConfig('2.0'),
+    ...createInitialReleaseConfig('1.3'),
+    ...createInitialReleaseConfig('1.2'),
+    ...createInitialReleaseConfig('1.1'),
+    ...createInitialReleaseConfig('1.0'),
+    ...createCurrentReleaseConfig('1.3'),
+    ...createCurrentReleaseConfig('2.0'),
```
Comment (booc0mtaco) [inline .storybook/preview.tsx:88]: These represent badges added to each component, showing when they were added, and what their current version was created. might also be used for metadata in omlet component analytics in the future.

File: bin/eds-apply-theme.js [hunk 2]
```diff
@@ -85,8 +64,6 @@
   });
 
   try {
-    // Keys in the theme file must be a strict subset of those in the base file
-    isStrictSubset(baseTheme, localTheme);
```
Comment (booc0mtaco) [inline bin/eds-apply-theme.js:89]: Themes can now introduce tokens that are local to the application, and aren't just a subset of what EDS provides. useful for adding custom colors, then mapping them to tier-2 tokens

File: .storybook/components/ColorList/ColorList.module.css [hunk 3]
```diff
@@ -1,7 +1,7 @@
 .color-list__label {
   width: fit-content;
-  padding: var(--eds-size-half);
-  border-radius: var(--eds-theme-box-button-border-radius);
+  padding: calc(var(--eds-size-half) / 16 * 1rem);
```
Comment (booc0mtaco) [inline .storybook/components/ColorList/ColorList.module.css:3]: Size tokens are now unit-less, to support importing from figma, AND to allow usage with tailwind to generate size utility classes (e.g., `m-*`, `p-*`, etc.)

File: .storybook/pages/WireframeDemo/GlobalStyles.module.css
Comment (booc0mtaco) [inline .storybook/pages/WireframeDemo/GlobalStyles.module.css:1]: Removing the wireframe demo and will lean on the theming documentation to illustrate how best to use EDS with alternative themes.

File: src/components/Accordion/Accordion.stories.tsx [hunk 4]
```diff
@@ -5,23 +5,25 @@ import { Accordion } from './Accordion';
...
 
 type Args = React.ComponentProps<typeof Accordion>;
 
 export default {
   title: 'Components/Accordion',
   component: Accordion,
   parameters: {
-    badges: ['1.2'],
+    layout: 'centered',
+    badges: ['intro-1.2', 'current-2.0'],
```
Comment (booc0mtaco) [inline src/components/Accordion/Accordion.stories.tsx:16]: This will denote what "version" of EDS a component was added in, and where it is now. Can also include other badges like BETA, etc.

File: src/components/Accordion/Accordion.tsx [hunk 5]
```diff
@@ -39,33 +37,47 @@ type AccordionProps = {
...
-  onOpen?: () => void;
+  subtitle?: ReactNode;
+  /**
+   * The title/heading of the component
+   */
+  title?: string;
+  /**
+   * Icon override for component's expand/collapse indicator.
+   *
+   * **Default is `"expand-more"`**.
```
Comment (booc0mtaco) [inline src/components/Accordion/Accordion.tsx:78]: TODO: align with new icon name

File: src/components/AppNotification/AppNotification.module.css
Comment (booc0mtaco) [inline src/components/AppNotification/AppNotification.module.css:1]: TODO: use size tokens in here

File: src/components/AppNotification/AppNotification.tsx [hunk 6]
```diff
@@ -0,0 +1,94 @@
+import clsx from 'clsx';
+import React from 'react';
+import Button from '../Button';
+import Text from '../Text';
+import styles from './AppNotification.module.css';
+
+export interface AppNotificationProps {
+  // Design API
```
Comment (booc0mtaco) [inline src/components/AppNotification/AppNotification.tsx:8]: Props under "Design API" denote things that should match what shows in figma (both prop name, and values). After this, any discrepancies will be easier to address in figma, to avoid a breaking change.

File: src/components/AppNotification/AppNotification.tsx [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+  /**
+   * Secondary text used to describe the notification in more detail
+   */
+  subTitle: React.ReactNode;
+  /**
+   * Treatment for component (whether it is dark on light text, or light on dark text)
+   */
+  color: 'dark' | 'light';
+
+  // Component API
```
Comment (booc0mtaco) [inline src/components/AppNotification/AppNotification.tsx:22]: "Component API" are things which are more programmatic/for engineers.

File: src/components/Checkbox/Checkbox.module.css [hunk 8]
```diff
@@ -2,12 +2,12 @@
...
 .checkbox {
   display: flex;
-  gap: var(--eds-size-1);
+  gap: calc(var(--eds-size-1) / 16 * 1rem);
 }
 
 .checkbox__input {
-  height: 18px;
-  width: 18px;
+  height: 1.125rem;
```
Comment (booc0mtaco) [inline src/components/Checkbox/Checkbox.module.css:9]: Not using a size token here, but making it consistent with the others which use `rem`s for sizing

File: src/components/Checkbox/Checkbox.module.css [hunk 9]
```diff
@@ -18,59 +18,137 @@
...
-
-  /* Clip this element in the shape of a checkbox. The element's background color will be visible
-  anywhere the svg path is solid. Whatever is behind the checkbox will show through wherever the
-  svg is transparent. See https://developer.mozilla.org/en-US/docs/Web/CSS/clip-path */
-  clip-path: path(
-    'M10.6 16.2L17.65 9.15L16.25 7.75L10.6 13.4L7.75 10.55L6.35 11.95L10.6 16.2ZM3 21V3H21V21H3Z'
-  );
+  color: var(--eds-theme-color-text-neutral-default-inverse);
+  content: '✓';
+  text-align: center;
```
Comment (booc0mtaco) [inline src/components/Checkbox/Checkbox.module.css:35]: Changing up the implementation a bit here to achieve a few things:

- use the OS-specific glyph for the mark
- allow for adding color to the check itself, the background, and the border
- allow for shaping the checked state and disabled state to have rounded borders (which was a but requiring two clip-paths

**Review by booc0mtaco — COMMENTED** [review]
Have a look at the built storybook to see how each one looks! Adding some comments for guidance and context on a few consistent changes throughout this PR.

**Comment by ahuth** [comment]
Is the rounding at the top left of the Storybook top bar intended?I couldn't decide if I thought it looked weird or not.

Screenshot 2024-06-20 at 1 05 25 PM

**Review by ahuth — APPROVED** [review]

File: .eslintrc [hunk 10]
```diff
@@ -25,6 +25,7 @@
           "**/*.config.{js,ts,mjs}",
           "**/*.setup.{js,ts}",
           ".storybook/**",
+          ".storybook/*.{ts,js}",
```
Comment (ahuth) [inline .eslintrc:28]: Why is this needed? I'd expect the line above to lint all files under .storybook/. If they're not, then the line above should probably be fixed or removed.

Comment (booc0mtaco) [reply .eslintrc:28]: I changed this a while back to fix something that was happening but can't remember what exactly. it's likely merging the two would be what we want, e.g., `".storybook/**/*"` or something. I can test that out to confirm

**Comment by booc0mtaco** [comment]
> Is the rounding at the top left of the Storybook top bar intended?I couldn't decide if I thought it looked weird or not.
> 
> Screenshot 2024-06-20 at 1 05 25 PM

Yah, this comes from the storybook theming config, which offers ONLY one radius to style

```
  appBorderRadius: 4, // border-radius-md
```

I can reset this to 0, which cleans up that corner (but also makes the Docs page example for each component also have a square corner radius... actually more consistent with how cards and such look in the new theme).

**Review by timzchang — APPROVED** [review]
